From d505ddbb7f46ba974b5d8458fd7134901930f033 Mon Sep 17 00:00:00 2001
From: Jan200101 <sentrycraft123@gmail.com>
Date: Thu, 17 Feb 2022 18:00:58 +0100
Subject: [PATCH] ignore target lib dirs when invoked with -feach-lib-rpath

Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
---
 src/main.zig | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/main.zig b/src/main.zig
index 7f50a7de6..a6bcf69a8 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -2077,10 +2077,6 @@ fn buildOutputType(
             clang_argv.appendAssumeCapacity(framework_dir);
             framework_dirs.appendAssumeCapacity(framework_dir);
         }
-
-        for (paths.lib_dirs.items) |lib_dir| {
-            try lib_dirs.append(lib_dir);
-        }
         for (paths.rpaths.items) |rpath| {
             try rpath_list.append(rpath);
         }
-- 
2.35.1

