From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aleksei Bavshin <alebastr89@gmail.org>
Date: Fri, 20 Oct 2023 04:49:54 +0000
Subject: [PATCH] Enable build-id by default

Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
---
 src/Compilation.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Compilation.zig b/src/Compilation.zig
index 507cbfc6d5..d946956b05 100644
--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -1230,7 +1230,7 @@ pub fn create(gpa: Allocator, arena: Allocator, options: CreateOptions) !*Compil
         const any_sanitize_thread = options.config.any_sanitize_thread or options.root_mod.sanitize_thread;
 
         const link_eh_frame_hdr = options.link_eh_frame_hdr or any_unwind_tables;
-        const build_id = options.build_id orelse .none;
+        const build_id = options.build_id orelse .sha1;
 
         const link_libc = options.config.link_libc;
 
